### ISSUE TITLE
create data sensitivity ontology

### ISSUE THREAD

**Issue by mbjones** [DataONEorg/sem-prov-ontologies#113]
Through our repositories, we have a need to consistently represent data sensitivity classifications for data, and using annotations against a data sensitivity ontology has been proposed. In this task, create an initial ontology that captures the data sensitivity categories for repositories and relates them to existing sensitivity taxonomies, including [DataTags](https://datatags.org), the [Data Use Ontology (DUO)](https://github.com/EBISPOT/DUO/), and the [Informed Consent Ontology (ICO)](https://github.com/ICO-ontology/ICO).

See also: Information about the categories we are using for the Arctic Data Center is in the [sensitive data matrix](https://docs.google.com/spreadsheets/d/19RxJLkq8g6mJwnkSaP6TT66J9Ympz27UVhL77_6LsvI).

Candidate names:
- Data Sensitivity Ontology (SENSO)
- ?

**Comment by mbjones** [comment]
Started this with an initial file in a branch `feature-113-create-senso` in SHA 4ae8a14

This small ontology creates the three classes we agreed on for Arctic Data Center for `Non-sensitive data`, `De-identified data`, and `Sensitive data`, with definitions. It also creates classes for the `DataTags` categories (https://datatags.org), and links those where they directly correspond. I also imported the DUO ontology and was looking at where there might be equivalences there, but I am currently thinking they might be complementary and not directly related. However, it is possible that DUO's unrestricted data permissions apply to the non-sensitive data categories, etc. It would be good to discuss the relationship between sensitivity, confidentiality, and data use permissions as concepts in these ontologies.

@jeanetteclark @laurenwalker @amoeba could you take a look for the ADC annotations for sensitive data as we have been mocking up? Thx.

**Comment by mbjones** [comment]
Issues to be resolved:

- [ ] add (or find) a property for attaching sensitivity classes to datasets in annotations
- [ ] Decide if Data Sensitivity Category (`odo:SENSO_00000001`) should be a subclass of  "data use modifier" (`obo:DUO_0000017`)
- [ ] decide on and create alignments between the ADC sensitivity classes and the DataTags classes
    - @mpsaloha suggests trying `skos-dl:exactMatch` and `skos-dl:closeMatch`
- [ ] Clarify the DataTags definitions and try to eliminate the identical definitions
- [ ] Decide if SENSO should be its own ontology or incorporated within another
    - this might be a "DataONE Annotation Ontology" or an Arctic Data Center Annotation Ontology"?
    -

**Comment by mbjones** [comment]
Added modifications to try using `owl:equivalentClass` for alignment, and it produced the expected inferred graph. Seems like a good way to proceed. Bryce is testing how this will work with the indexer and search.

**Comment by amoeba** [comment]
We had a great discussion on our salmantics call about the above points. For modeling and alignment issues, we decided to skip them for now and be open to revisiting them later. The first version of SENSO will just have three sub-classes and won't include terms for the DataTags or alignments between them. Not making the assertions now doesn't preclude us from making them in the future but making assertions now that we later on might change might cause a lot of trouble.

On the last issue,

>  [ ] Decide if SENSO should be its own ontology or incorporated within another

Some points were brought up:

- Publishing a topic ontology such as SENSO kinda implies it might cover the topic comprehensively, be usable to others, might be expanded, and might take contributions from others. The set of terms we have now is almost certainly not covering everyone's mental model of data sensitivity.
- If we continue making these one-off (bespoke) ontologies, will that impact building user interfaces negatively?
  
  The only issue that really stands out here is that we don't want to be displaying bewilderingly long lists of choices to users in either search or editing user interfaces. Using lots of modular ontologies does present a challenge there. But that's not really issue we have now and maybe not one we will have in the future. If we do get to that point, we could create application-specific ontologies (ie. an ontology just for the Arctic Data Center) which _import_ some of these modular ones, effectively reducing the total list of ontologies we might display to a user.

Ultimately, we agreed to go with the modular approach and continue down the path of building SENSO out as mentioned above.

I'll make these changes and ping others for a second look.

**Comment by amoeba** [comment]
I made the edits we talked about (see commit message in b694039a91ab554e2c1b6f82cbea95042c1d7441). One thing we hadn't caught that I changed was that the terms were using URIs like http://purl.obolibrary.org/odo/SENSO_ instead of http://purl.dataone.org/odo/SENSO_. I opted to change them all to http://purl.dataone.org/odo/SENSO_ since that matches the ontology IRI.

@mbjones @mpsaloha please take a look.

**Comment by mbjones** [comment]
Looks good, @amoeba. I pushed another set of edits that fixed the `odo` prefix definition, and reformatted subjects to use short form in the TTL file. I checked off the boxes above on which I think we have made decisions. A few items remaining, but I think they should probably be opened as new tickets. So, what do we need to add to add SENSO into the website listing too?

**Comment by amoeba** [comment]
Thanks @mbjones. New tickets sounds good, I can file those in a sec.

I'll add a commit on top of this branch here that'll enable it on the GitHub pages site when we merge to _main_. Once merged, I think we should get this off to BioPortal so landing page popovers can pull in definitions.

**Comment by amoeba** [comment]
Okay, that's all done. IMO this is ready for a merge to develop and a PR onto main after that. If you just want to give me a thumbs up I'll get those going.

**Comment by amoeba** [comment]
Hey @mbjones, I'm ready to move this forward into production but I did want a final 👍 from you before I did that. Let me know.

**Comment by jeanetteclark** [comment]
@mbjones @amoeba sorry - missed the review here. I just had a look and looks good to me

**Comment by amoeba** [comment]
Great, thanks @mbjones, @jeanetteclark.

**Comment by amoeba** [comment]
- SENSO is merged to `main`: https://github.com/DataONEorg/sem-prov-ontologies/pull/111
- Tagged + released: https://github.com/DataONEorg/sem-prov-ontologies/releases/tag/arcrc-0.10-senso-0.1.0
- On its way to BioPortal: https://bioportal.bioontology.org/ontologies/SENSO

Next/last steps are to include it in our DataONE and Metacat index components. Marching off to file those tickets now.

**Comment by laurenwalker** [comment]
Hey @amoeba - I am getting a 404 for the purl.dataone.org links above. Are they not production ready yet?

**Comment by amoeba** [comment]
Not quite. I can pick up the remaining work on that, it was just at the bottom of my todo list.

**Comment by amoeba** [comment]
Okay @laurenwalker, SENSO redirects are in place so those URIs should also resolve now.

**Comment by laurenwalker** [comment]
> Okay @laurenwalker, SENSO redirects are in place so those URIs should also resolve now.

Great, thanks Bryce!
